gomatrixserverlib/pipeline.yml

```diff
@@ -0,0 +1,21 @@
+steps:
+  - command:
+      - "env"
+      - "go test ./..."
+    label: "\U0001F9EA Unit tests / :go: 1.13"
+    plugins:
+      - docker#v3.0.1:
+          image: "golang:1.13"
+          mount-buildkite-agent: false
+
+  - command:
+      - "env"
+      # https://github.com/golangci/golangci-lint#memory-usage-of-golangci-lint
+      - "golangci-lint run"
+    label: "\U0001F9F9 Lint / :go: 1.13"
+    agents:
+      queue: "xlarge"
```

Comment:
REALLY?

Reply:
This is the size that's used by dendrite's linter. Are these sizes documented anywhere?

Reply:
Looking at the agents in BK it seems these are actually based on EC2 T3 sizes, so yes, 16GB does seem like a lot. `golangci-lint` is quite memory heavy but I'll put it down to `medium` (4GB RAM) for now.

Reply:
supposedly the queues are documented at https://gitlab.matrix.org/new-vector/internal/-/wikis/BuildKite#build-queues, but they are a bit light on detail...

Reply:
(apart from saving us money, an advantage of using a smaller instance is that you are more likely to find a spare agent sitting around idle rather than having to wait while one gets spun up - or sitting in the queue, since we limit the number of xlarge agents more restrictively.)